### PR TITLE
Cleanup firewalls, add nginx ingress to presubmit

### PR DESCRIPTION
**Pull request by bprashanth**
Make the firewall cleanup code follow the same pattern as the other cleanup functions, and add the nginx ingress e2e to presubmit. 

Planning to watch the test for a bit, and if it works alright, I'll add the other Ingress e2e to post-submit merge blocker.

### CODE REVIEW

**Comment by bprashanth** [comment]
@k8s-bot gce etcd3 e2e test this

File: test/e2e/ingress.go [hunk 1]
```diff
@@ -151,7 +151,7 @@ var _ = framework.KubeDescribe("Loadbalancing: L7 [Feature:Ingress]", func() {
 	})
 
 	// Time: borderline 5m, slow by design
-	framework.KubeDescribe("Nginx [Slow]", func() {
+	framework.KubeDescribe("Nginx", func() {
```
Comment (spxtr) [inline test/e2e/ingress.go]: The last run in this PR took over 5 minutes. Can this be sped up a bit?

**Comment by bprashanth** [comment]
Usually on my local machine it takes ~3.50, and that's because it's running > 1 test within a single "It" block: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/ingress.go#L192. we do it this way because we need common cloud resources for all the conformance tests, that we setup/teardown in Before and After each, so in theory this is like 4 tests. 

I agree if it took more than 6m on the builder we can probably think of something (most simply, just run a subset of the conformance suite, but maybe optimize). Can you point me at the logs of the long run? loading each log through the web ui seems to be taking a really long time.

**Comment by spxtr** [comment]
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/38141/pull-kubernetes-e2e-gce/8354/build-log.txt

```
I1205 14:33:01.957] [32mâ€¢ [SLOW TEST:321.647 seconds][0m
I1205 14:33:01.957] [k8s.io] Loadbalancing: L7
I1205 14:33:01.957] [90m/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:826[0m
I1205 14:33:01.957]   [k8s.io] Nginx
I1205 14:33:01.957]   [90m/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:826[0m
I1205 14:33:01.957]     should conform to Ingress spec
I1205 14:33:01.957]     [90m/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/ingress.go:198[0m
```

(it would be nice if I could see passing tests on Gubernator without looking in the build log, @rmmh)

**Comment by bprashanth** [comment]
This one should run faster

**Comment by bprashanth** [comment]
@spxtr PTAL
```
I1206 14:55:28.761] 
I1206 14:55:28.761] [32m• [SLOW TEST:227.813 seconds][0m
I1206 14:55:28.761] [k8s.io] Loadbalancing: L7
I1206 14:55:28.761] [90m/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:826[0m
I1206 14:55:28.761]   [k8s.io] Nginx
I1206 14:55:28.762]   [90m/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:826[0m
I1206 14:55:28.762]     should conform to Ingress spec
I1206 14:55:28.762]     [90m/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/ingress.go:201[0m
I1206 14:55:28.762] [90m------------------------------[0m
```

I'm still not able to open the full build-log in my browser, I need to gsutil cp. Just me or is something else up?

**Pull request merged**